### ISSUE TITLE
GROUP BY requires aggregate functions even if all expressions are already aggregated

### ISSUE THREAD

**Issue by znxftw** [confluentinc/ksql#8953]
(ksql 0.23.1) Having a GROUP BY clause in a SELECT query forces you to have at least one aggregate function in the query, even if all the expressions have been aggregated.

When doing
```sql
CREATE TABLE T(A INT PRIMARY KEY, B INT) WITH(KAFKA_TOPIC='T',VALUE_FORMAT='JSON',PARTITIONS=1);
```

```sql
SELECT B FROM T GROUP BY B EMIT CHANGES;
```
gives `GROUP BY requires aggregate functions in either the SELECT or HAVING clause`, but
```sql
SELECT B, COUNT(*) FROM T GROUP BY B EMIT CHANGES;
```
works fine.

Even though the expressions in the `SELECT` statement do not have any ambiguity due to grouping, KSQL forces you to add a dummy aggregated column to be able to execute the above query.

Was there any reason for enforcing the requirement of an aggregated expression when doing a `GROUP BY`? Logically, the above expression should be able to run without any ambiguity.

**Expected behavior**
If there are no un-aggregated expressions in an otherwise valid SELECT query without any aggregate functions, it should execute successfully.

**Actual behaviour**
If there are no un-aggregated expressions in an otherwise valid SELECT query without any aggregate functions, we get the following error : `GROUP BY requires aggregate functions in either the SELECT or HAVING clause`

**Comment by ayushkhandelwal317** [comment]
+1

**Comment by mjsax** [comment]
> Was there any reason for enforcing the requirement of an aggregated expression when doing a GROUP BY? Logically, the above expression should be able to run without any ambiguity.

Yes, for this case following standard SQL is tricky. The underlying Kafka key-value model is the "issue" here. All grouping columns become the key in the result key-value-pairs -- and if there is no value-columns we would get a `<key,null>` what is a tombstone (ie, a delete) in Kafka.

Maybe we could insert some "dummy byte" into the value for this case to make it work? Not sure if this would be an acceptable solution?

**Comment by znxftw** [comment]
Ah yes, that makes sense. I'm guessing this issue indirectly relates to #5564 then.

**Comment by znxftw** [comment]
> Yes, for this case following standard SQL is tricky. The underlying Kafka key-value model is the "issue" here. All grouping columns become the key in the result key-value-pairs -- and if there is no value-columns we would get a `<key,null>` what is a tombstone (ie, a delete) in Kafka.

A small follow up to this,

Even if I do 
```sql
SELECT B, -1 as DUMMY FROM T GROUP BY B EMIT CHANGES;
```
it gives the same error, `GROUP BY requires aggregate functions in either the SELECT or HAVING clause`. 

In this case, `DUMMY` would be a value column in the push query wouldn't it? Why would ksql still force me to have an aggregate function?

**Comment by znxftw** [comment]
Awesome, thanks!
